src/main/sensors/compass.c

```diff
@@ -73,15 +73,18 @@ void updateCompass(flightDynamicsTrims_t *magZero)
 {
     static uint32_t calStartedAt = 0;
     static int16_t magPrev[XYZ_AXIS_COUNT];
-    uint32_t axis;
 
-    mag.read(magADCRaw);
-    for (axis = 0; axis < XYZ_AXIS_COUNT; axis++) magADC[axis] = magADCRaw[axis];  // int32_t copy to work with
+    if (!mag.read(magADCRaw)) {
+        return;
```

Comment:
We still need to return zero vector here so IMU can detect invalid compass and act accordingly.